sklearn/feature_selection/_univariate_selection.py

```diff
@@ -294,15 +294,34 @@ def f_regression(X, y, center=True):
     else:
         X_norms = row_norms(X.T)
 
+    # the Pearson correlation coefficient is undefined
+    # for random variables (regressors and target) with
+    # a standard deviation of zero
+    if np.any(X_norms == 0) or np.std(y) == 0:
+        raise ValueError("The correlation is undefined for "
+                         "regressors and target with "
+                         "identically constant values. "
+                         "Remove them before calling this method.")
+
     # compute the correlation
     corr = safe_sparse_dot(y, X)
     corr /= X_norms
     corr /= np.linalg.norm(y)
 
-    # convert to p-value
+    # compute degrees of freedom
```

Comment:
I'd remove this comment, it's quite obvious what's done in the next line

Reply:
You are absolutely right, let us keep things simpler and clearer.

Fixed!